osquery/tables/utility/file.cpp

```diff
@@ -99,6 +127,171 @@ void genFileInfo(const fs::path& path,
     r["type"] = "unknown";
   }
 
+#else
+
+  // Get the handle of the file object.
+  hFile = CreateFile(TEXT(path.string().c_str()),
+                     GENERIC_READ,
+                     FILE_SHARE_READ,
+                     NULL,
+                     OPEN_EXISTING,
+                     FILE_ATTRIBUTE_NORMAL,
+                     NULL);
+
+  // Check GetLastError for CreateFile error code.
+  if (hFile == INVALID_HANDLE_VALUE) {
+    return;
+  }
+
+  // Get the owner SID of the file.
+  dwRtnCode = GetSecurityInfo(hFile,
+                              SE_FILE_OBJECT,
+                              OWNER_SECURITY_INFORMATION,
+                              &pSidOwner,
+                              NULL,
+                              NULL,
+                              NULL,
+                              &pSD);
+
+  // Check GetLastError for GetSecurityInfo error condition.
+  if (dwRtnCode != ERROR_SUCCESS) {
+    return;
+  }
+
+  FILE_BASIC_INFO basicInfo;
+  BY_HANDLE_FILE_INFORMATION fileInfo;
+
+  if (0 == GetFileInformationByHandle(hFile, &fileInfo)) {
+    return;
+  }
+
+  auto fileIndex = ((unsigned long long)fileInfo.nFileIndexHigh << 32) +
+                   (unsigned long long)fileInfo.nFileIndexLow;
+
+  std::stringstream stream;
+  stream << "0x" << std::setfill('0')
+         << std::setw(sizeof(unsigned long long) * 2) << std::hex << fileIndex;
+  std::string fileId(stream.str());
+
+  // Windows has fileid's that are displayed in hex using: 
+  // fsutil file queryfileid <filename>
+  r["fileId"] = TEXT(fileId);
+
+  // inode is the decimal equivalent of fileid
+  r["inode"] = BIGINT(fileIndex);
+  r["uid"] = INTEGER(getUidFromSid(pSidOwner));
+  r["gid"] = INTEGER(getGidFromSid(pSidOwner));
+
+  // Permission bits don't make sense for Windows. Use ntfs_acl_permissions
+  // table
+  r["mode"] = "-1";
```

Comment:
I agree that the modes don't make much sense, but we do make some effort to derive a "octet" of perms. That being said, we also have an `ntfs_acl` perms table that's on the way, so I'd say let's leave this as it was.

Reply:
Just so I'm clear do you still want to use stat to generate an octet? If so what user permissions was it previously pulling? Was it the account permissions the process was running as?

Reply:
I think is one of the last items? ^